gdal_subsetter/transform.py

```diff
@@ -1405,29 +1410,26 @@ def _process_geogcs(ds_in, dst):
                 if gt[2] == 0.0 and gt[4] == 0.0:
                     datavar.coordinates = 'lon lat'
 
-        # open infile
-        ds_in = gdal.Open(infile)
-        # create a output nc file
-        dst = Dataset(outfile, mode='w', format='NETCDF4')
-        # define global attributes
-        dst.title = ''
-        dst.institution = 'Alaska Satellite Facility'
-        dst.source = ''
-        dst.references = ''
-        dst.comment = ''
-        dst.history = f'{datetime.utcnow():%d/%m/%Y %H:%M:%S} (UTC)'
-        dst.GDAL = f'Version {gdal.__version__}'
-
-        # create dimensions
-        crs = parse_crs_from_ogc_wkt(ds_in.GetProjectionRef())
-        if crs.cs_type == 'Projected':
-            _process_projected(ds_in, dst)
-        else:
-            _process_geogcs(ds_in, dst)
+        with OpenGDAL(infile) as input_geotiff, \
+            Dataset(outfile, mode='w', format='NETCDF4') as output_netcdf4:
```

Comment:
I'm looking forward to this service using Python 3.10 - [one of the new features](https://docs.python.org/3/whatsnew/3.10.html) is the ability to use parentheses with context managers.

Reply:
Can we do that now?  (upgrade to 3.10, we control the service image right?)

Reply:
The base image is `osgeo/gdal:ubuntu-full-3.4.3`, rather than just a Python image. I'll do some research to see if any of the `osgeo/gdal` images support Python 3.10. That feels like it has the potential to maybe break things, so I'll do that separately (it's funny, I thought code clean up would reduce the list of things to do, but it just keeps turning up more)

Reply:
Oh yeah, bummer.

Reply:
So, possibly updating to `osgeo/gdal:ubuntu-full-3.6.0` could get us to Python 3.10 (if I'm reading [this line](https://github.com/OSGeo/gdal/blob/v3.6.0/docker/ubuntu-full/Dockerfile#L291) correctly). I'll give that a go in one of the next few PRs for this service.